src/vivarium_gates_nutrition_optimization/components/observers.py

```diff
@@ -220,21 +222,48 @@ def setup(self, builder: Builder) -> None:
                 excluded_stratifications=self.config.exclude,
             )
 
+
 class DisabilityObserver(DisabilityObserver_):
     def setup(self, builder: Builder) -> None:
-        super().setup(builder)
+        self.config = builder.configuration.stratification.disability
+        self.step_size = pd.Timedelta(days=builder.configuration.time.step_size)
+        self.disability_weight = self.get_disability_weight_pipeline(builder)
+        cause_states = builder.components.get_components_by_type(
+            tuple(self.disease_classes)
+            # Hack in Anemia
+        ) + [State("anemia")]
+        base_query = 'tracked == True and alive == "alive"'
+
         builder.results.register_observation(
-            name=f"ylds_due_to_anemia",
-            pop_filter='tracked == True and alive == "alive"',
-            aggregator_sources=["anemia.disability_weight"],
+            name="ylds_due_to_all_causes",
+            pop_filter=base_query,
+            aggregator_sources=[self.disability_weight_pipeline_name],
             aggregator=self._disability_weight_aggregator,
             requires_columns=["alive"],
-            requires_values=["anemia.disability_weight"],
+            requires_values=["disability_weight"],
             additional_stratifications=self.config.include,
             excluded_stratifications=self.config.exclude,
             when="time_step__prepare",
         )
 
+        for cause_state in cause_states:
+            cause_disability_weight_pipeline_name = (
+                f"{cause_state.state_id}.disability_weight"
+            )
+            builder.results.register_observation(
+                name=f"ylds_due_to_{cause_state.state_id}",
+                pop_filter=base_query
+                if cause_state.state_id == "maternal_disorders"
```

Comment:
Why aren't maternal disorders paused during parturition state?

Reply:
We want Maternal disorders to be the only thing that is counted during the parturition state (maternal disorders can only exist during the parturition state); we don't want to simultaneously include any other source of disability because the maternal disorder YLDs are already adjusted for comorbidity.